### DISCUSSION TITLE
ViteJS counterpart in Deno

### DISCUSSION THREAD

**Discussion by ejsmith** [denoland/deno#13419] [Ideas]
Wouldn't it be awesome to be able to write code that looks exactly like any other Deno code but for the browser? ViteJS provides a loader that intercepts requests and transpiles files on the fly pretty much the same thing Deno does transpiling typescript files on the fly. It would be an incredible dev experience to be able to use the Deno tooling and have a Deno web server that works like the vite dev server does. Thoughts?

**Comment by bartlomieju** [comment]
We are actively working on being able to run `vite` directly in Deno

**Reply by ejsmith** [reply]
Would that allow the code you are writing for the client to look exactly like the Deno server side code?

I'm currently having a horrible time trying to get a vite client side app to work together with a Deno server. It seems like the multi-root workspaces support where one folder has deno enabled and the other doesn't is not working at all.

**Reply by bartlomieju** [reply]
@ejsmith are you talking about having the same module resolution algorithm as Deno uses? If so I believe this will be possible, as Vite allows to configure the resolution plugin (AFAIK).

**Reply by itsdouges** [reply]
Does a plugin exist for enabling Vite to use the Deno module resolution atm?
